File: src/main/java/io/eiren/vr/trackers/TrackerStatus.java
package io.eiren.vr.trackers;

public enum TrackerStatus {
	
	DISCONNECTED(0, false),
	OK(1, true),
	BUSY(2, true),
	ERROR(3, false),
	OCCLUDED(4, false),
	;
	
	private static final TrackerStatus byId[] = new TrackerStatus[5];
	
	public final int id;
	public final boolean sendData;
	
	private TrackerStatus(int id, boolean sendData) {
		this.sendData = sendData;
		this.id = id;
	}
	
	public static TrackerStatus getById(int id) {
		if(id < 0 || id >= byId.length)
			return null;
		return byId[id];
	}
	
	static {
		for(TrackerStatus st : values())
			byId[st.id] = st;
	}
}

File: server/core/src/main/java/dev/slimevr/tracking/trackers/TrackerStatus.kt
package dev.slimevr.tracking.trackers

enum class TrackerStatus(val id: UByte, val sendData: Boolean, val reset: Boolean) {

	DISCONNECTED(0u, false, true),
	OK(1u, true, false),
	BUSY(2u, true, false),
	ERROR(3u, false, true),
	OCCLUDED(4u, false, false),
	TIMED_OUT(5u, false, false),
	;

	companion object {

		private val byId = entries.associateBy { it.id }

		@JvmStatic
		fun getById(id: UByte): TrackerStatus? = byId[id]
	}
}
